.github/workflows/pylint_checking.yml

```
@@ -0,0 +1,44 @@
+---
+name: Pylint
+on: pull_request
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9]
+    steps:
+      - uses: actions/checkout@v1
```

is there a reason for using v1 of the both the checkout and python actions
```suggestion
      - uses: actions/checkout@v4
```

---

I try to use the same as the "formating.yml". However, I think we can change to v4